File: ruby/resistor-color-duo/resistor_color_duo.rb
=begin
  This passes all the given test conditions, but something
  like ResistorColorDuo.value(["black", "brown"]) will return
  1 instead of 01.
=end

module ResistorColorDuo
  COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]

  def self.value array
    (COLORS.index(array[0]).to_s + COLORS.index(array[1]).to_s).to_i
  end
end

File: typescript/resistor-color-duo/resistor-color-duo.ts
const colorArray = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
] as const;

export type Color = typeof colorArray[number];

export function decodedValue([c1, c2]: Color[]): number {
  return colorArray.indexOf(c1) * 10 + colorArray.indexOf(c2);
}
